flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -781,6 +787,32 @@ class TaskManager(
     }
   }
 
+  private def handleRequestTaskManagerLog
+  (sender: ActorRef, requestType: LogTypeRequest, jobManager: ActorRef):
+  Unit = {
+    if (blobService.isDefined && jobManager.path.address.host.isDefined) {
+      //create new Thread to upload log, to not block the TM for too long
+      new Thread() {
+        override def run(): Unit = {
+          val jmHost = jobManager.path.address.host.get
+          val port = blobService.get.getPort
+          val logFilePath = System.getProperty("log.file");
+          val file: File = requestType match {
+            case LogFileRequest => new File(logFilePath);
+            case StdOutFileRequest =>
+              new File(logFilePath.substring(0, logFilePath.length - 4) + ".out");
+          }
+
+          val client: BlobClient = new BlobClient(new InetSocketAddress(jmHost, port))
+
+          sender ! client.put(new FileInputStream(file))
+        }
+      }.run()
+    } else {
+      throw new IOException("No BlobService defined, cannot upload TaskManager logs.")
```

Comment:
Throwing this exception here will terminate the TM. I think the exception should rather be send to the `sender` where it is logged.